File: WangLulu/Stack/EvaluateReversePolishNotations.java
package Stack;

import java.util.Stack;

public class EvaluateReversePolishNotations {
     public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        for (String s : tokens) {
            if (isNumber(s)) {
                stack.push(Integer.parseInt(s));
            } else {
                int i = stack.pop();
                int j = stack.pop();
                switch (s) {
                    case "+":
                        stack.push(j + i);
                        break;
                    case "-":
                         stack.push(j-i);
                         break;
                    case "*":
                        stack.push(j*i);
                        break;
                    case "/":
                        stack.push(j/i);
                        break;
                }
            }
        }
        return stack.pop();
    }
    public boolean isNumber(String s){
        if(s.equals("+")||s.equals("-")||s.equals("*")||s.equals("/")){
            return false;
        }else{
            return true;
        }
    }
    
}

File: WangBo/栈&DFS/Evaluate Reverse Polish Notations.cpp
#include <iostream>
#include <vector>
#include <stack>
#include <string>
using namespace std;


class Solution {
    public:
        int evalRPN(vector<string>& tokens) {
            stack<string> st;
            int a;
            int b;
            for (size_t i = 0; i < tokens.size(); i++)
            {
                if (tokens[i]== "+" || tokens[i]== "-"|| tokens[i]== "*"|| tokens[i]== "/")
                {
                    b = stoi(st.top());
                    st.pop();
                    a = stoi(st.top());
                    st.pop();
                    if (tokens[i]== "+")
                    {
                        st.push(to_string(a+b));
                    }
                    else if (tokens[i]== "-")
                    {
                        st.push(to_string(a-b));
                    }
                    else if (tokens[i]== "*")
                    {
                        st.push(to_string(a*b));
                    }
                    else if (tokens[i]== "/")
                    {
                        st.push(to_string(a/b));
                    }   
                }
                else
                {
                    st.push(tokens[i]);
                }
            }
            return stoi(st.top());
        }
    };